src/ur5e_arm.cpp

```diff
@@ -579,6 +586,68 @@ void write_joint_pos_rad(vector6d_t js, std::ostream& of, unsigned long long uni
 
 // helper function to read a data packet and send a noop message
 bool UR5eArm::read_joint_keep_alive(bool log) {
+    // check to see if an estop has occurred.
+    std::string status;
+
+    if (!dashboard->commandSafetyStatus(status)) {
+        // we currently do not attempt to reconnect to the dashboard client. hopefully this error resolves itself.
+        BOOST_LOG_TRIVIAL(error) << "read_joint_keep_alive dashboard->commandSafetyStatus() returned false when retrieving the status";
+        return false;
+    }
+
+    if (status.find(urcl::safetyStatusString(urcl::SafetyStatus::NORMAL)) == std::string::npos) {
+        // the arm is currently estopped. save this state.
+        estop.store(true);
+
+        // clear any currently running trajectory.
+        trajectory_running.store(false);
+
+        // sleep longer to prevent buffer error
+        // Removing this will cause the RTDE client to move into an unrecoverable state
+        usleep(ESTOP_DELAY);
```

Comment:
Haven’t found the reason for why we need this. All I know is the arm moves into an unrecoverable state without this sleep, getting slammed with buffer errors(which doesn’t make a ton of sense)